### ISSUE TITLE
Crash on postprocessing No. 2

### ISSUE THREAD

**Issue by ChWick** [OMMR4all/ommr4all-line-detection#2]
In GitLab by @chwick on Mar 16, 2019, 22:15

Maybe you have a clue what could be the reason for this crash. If not I try to find and prepare the required data.
```
Traceback (most recent call last):
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 198, in <module>
    experimenter.run(args)
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 150, in run
    counts, metrics = zip(*map(run_single, train_args))
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 87, in run_single
    counts = evaluator.evaluate(args.test_pcgts_files)
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/evaluator.py", line 14, in evaluate
    for p in self.predictor.predict(pcgts_files):
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/pixelclassifier/predictor.py", line 57, in predict
    predictions = list(self.line_detection.detect(gray_images))
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 109, in detect_advanced
    yield self.detect_staff_lines(data[i])
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 138, in detect_staff_lines
    staff_list = self.postprocess_staff_systems(staff_list, staff_line_height, binarize(image_data.image))
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetector.py", line 299, in postprocess_staff_systems
    if image[st_point][i] != 0:
IndexError: index 72 is out of bounds for axis 0 with size 72
```

**Comment by ChWick** [comment]
In GitLab by @chwick on Mar 16, 2019, 22:57

I even got one with a negative index:
```
Traceback (most recent call last):
  File "omr/stafflines/detection/experimenter.py", line 205, in <module>
    experimenter.run(args)
  File "omr/stafflines/detection/experimenter.py", line 155, in run
    counts, metrics = zip(*map(run_single, train_args))
  File "omr/stafflines/detection/experimenter.py", line 91, in run_single
    counts = evaluator.evaluate(args.test_pcgts_files)
  File "/autofs/ls6/wick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/evaluator.py", line 14, in evaluate
    for p in self.predictor.predict(pcgts_files):
  File "/autofs/ls6/wick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/pixelclassifier/predictor.py", line 58, in predict
    for i, (data, r) in enumerate(zip(dataset.data, predictions)):
  File "/home/ls6/wick/venvs/ommr4all/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 109, in detect_advanced
    yield self.detect_staff_lines(data[i])
  File "/home/ls6/wick/venvs/ommr4all/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 138, in detect_staff_lines
    staff_list = self.postprocess_staff_systems(staff_list, staff_line_height, binarize(image_data.image))
  File "/home/ls6/wick/venvs/ommr4all/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetector.py", line 299, in postprocess_staff_systems
    if image[st_point][i] != 0:
```

**Comment by ChWick** [comment]
In GitLab by @chwick on Mar 17, 2019, 10:43

![bug](/uploads/8ff79cf642ee1722b4ebac7f82de329f/bug.png)

Leads to following crash, no model was used!
```
Traceback (most recent call last):
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/pixelclassifier/predictor.py", line 93, in <module>
    for prediction in detector.predict(pcgts):
  File "/home/cwick/Documents/Projects/ommr4all/modules/ommr4all-server/omr/stafflines/detection/pixelclassifier/predictor.py", line 63, in predict
    for i, (data, r) in enumerate(zip(dataset.data, predictions)):
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 98, in detect_basic
    yield self.detect_staff_lines(image_data)
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 138, in detect_staff_lines
    staff_list = self.postprocess_staff_systems(staff_list, staff_line_height, binarize(image_data.image))
  File "/home/cwick/Documents/Projects/ommr4all/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetector.py", line 299, in postprocess_staff_systems
    if image[st_point][i] != 0:
IndexError: index 82 is out of bounds for axis 0 with size 80
```

**Comment by ChWick** [comment]
In GitLab by @s331267 on Mar 17, 2019, 11:49

Fixed with 21fc8a516d74290f2ded538c8af70179091d7953

**Comment by ChWick** [comment]
In GitLab by @s331267 on Mar 17, 2019, 11:49

closed
